docs/key-manager/server-key-manager.md

```diff
@@ -0,0 +1,75 @@
+Do note that if you have a key manager passphrase or a BIP 39 passphrase set for a given seed, you will need these to be
+able to recover the seed.
+
+### Using multiple seeds
+
+You can change which wallet and key manager you are using by setting the `bitcoin-s.wallet.walletName` config
+option. If it is not set, it will use the default wallet. Different seeds are saved in the `seeds` folder, and the
+seed's file name will be prefixed with the wallet name.
+
+Currently, Bitcoin-S only supports using a single key manger/wallet at a single instance. When switching wallets, you
+will need to shut down the server and restart it with your modified config.
+
+### Passphrases
+
+Upon creating a wallet it's passphrase and the BIP 39 passphrase will be set to use what is in
+your [configuration file](../config/configuration.md). The key manager's passphrase is set using
+the `bitcoin-s.keymanager.aespassword` option, and the BIP 39 passphrase is set using
+the `bitcoin-s.keymanager.bip39password` option.
+
+If you do not have the set the `aespassword` config option on the creation of the key manager, then the seed will be
+saved unencrypted.
+
+#### Changing Key Manager Passphrase
+
+You can set and change your key mananger's passphrase using the `keymanagerpassphraseset`
+or `keymanagerpassphrasechange` rpc commands. For more details checkout out
+the [server docs](../applications/server.md#wallet).
+
+You cannot change your BIP39 passphrase because this would change the underlying keys. To change this you will need to
+create a new wallet and then send all the UTXOs from the previous wallet to the new one.
+
+### Restoring a Seed
+
+#### Restore from Mnemonic
+
+To restore from a mnemonic seed you can use the `importseed` cli command.
+It takes in the actual mnemonic seed, wallet name, and an optional passphrase to encrypt the seed.
+
+**Example:**
+
+```
+bitcoin-s-cli importseed "never gonna give you up never gonna let you down never gonna run around and desert you never gonna make you cry never gonna" "rickRollWallet" "insecurePassword"
+```
+
+#### Restore from XPRV
+
+To restore from a mnemonic seed you can use the `importxprv` cli command.
+It takes in the base58 encoded xprv, wallet name, and an optional passphrase to encrypt the seed.
+
+**Example:**
+
+```
+bitcoin-s-cli importxprv "xprv9uHRZZhk6KAJC1avXpDAp4MDc3sQKNxDiPvvkX8Br5ngLNv1TxvUxt4cV1rGL5hj6KCesnDYUhd7oWgT11eZG7XnxHrnYeSvkzY7d2bhkJ7" "xprvWallet" "insecurePassword"
+```
+
+#### Restore from backup
+
+To restore from a backed up seed, simply put the seed file in the `seeds` folder and then load the wallet.
+To do this, you will need the seed file name to be `walletName-encrypted-bitcoin-s-seed.json` or `encrypted-bitcoin-s-seed.json`
+if you want it to be the default wallet.
+To load it, just set the `bitcoin-s.wallet.walletName` config option to the appropriate wallet name.
+
+You made need to run a wallet rescan to restore funds, this can be done by using the `rescan` RPC command.
```

Comment:
So what's your opinion on using scala code examples vs cli examples?

Ideally we would have both, but it seems you favor the cli. I like scala code examples personally, and we get the added benefit of compile time check on the docs so we have to make sure we keep them updated. 

These are pretty important commands, so it is worth duplicating in my opinion. You have a pretty good narrative going on in the docs with the example too.

Reply:
Oh, i think I understand what you mean by "Server KeyManager" now -- this is using the key manager as part of the app server binary? 

So in this case, i think we should add the scala code examples above to the normal key manager doc then?